examples/plot_data_augmentation.py

```diff
@@ -0,0 +1,232 @@
+# Parameters for exponential moving standardization
+factor_new = 1e-3
+init_block_size = 1000
+
+preprocessors = [
+    Preprocessor('pick_types', eeg=True, meg=False, stim=False),  # Keep EEG sensors
+    Preprocessor(lambda x: x * 1e6),  # Convert from V to uV
+    Preprocessor('filter', l_freq=low_cut_hz, h_freq=high_cut_hz),  # Bandpass filter
+    Preprocessor(exponential_moving_standardize,  # Exponential moving standardization
+                 factor_new=factor_new, init_block_size=init_block_size)
+]
+
+preprocess(dataset, preprocessors)
+
+######################################################################
+# Cut Compute Windows
+# ~~~~~~~~~~~~~~~~~~~
+#
+
+from braindecode.preprocessing.windowers import create_windows_from_events
+
+trial_start_offset_seconds = -0.5
+# Extract sampling frequency, check that they are same in all datasets
+sfreq = dataset.datasets[0].raw.info['sfreq']
+assert all([ds.raw.info['sfreq'] == sfreq for ds in dataset.datasets])
+# Calculate the trial start offset in samples.
+trial_start_offset_samples = int(trial_start_offset_seconds * sfreq)
+
+# Create windows using braindecode function for this. It needs parameters to define how
+# trials should be used.
+windows_dataset = create_windows_from_events(
+    dataset,
+    trial_start_offset_samples=trial_start_offset_samples,
+    trial_stop_offset_samples=0,
+    preload=True,
+)
+
+######################################################################
+# Split dataset into train and valid
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+splitted = windows_dataset.split('session')
+train_set = splitted['session_T']
+valid_set = splitted['session_E']
+
+######################################################################
+# Defining a transform
+# ~~~~~~~~~~~~~~~~~~~~
+#
+
+######################################################################
+# Data can be manipulated by transforms. A transform is usually handled by a
+# custom data loader, but can also be called using the forward function. Here I
+# will use the forward function to get manipulated data for visualization.
+#
+
+# First, we need to define a transform.
+
+from braindecode.augmentation import FrequencyShift
+transform = FrequencyShift(
+    probability=1.,  # defines the probability by which a sample is manipulated
+    sfreq=sfreq,
+    delta_freq_range=(10., 10.)  # -> fixed frequency shift for visualization
+)
+
+######################################################################
+# Manipulating one session and visualizing the transformation
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+
+# Next, I will manipulate one session to show the resulting frequency shift. I
+# will manipulate the data of an mne Epoch to make usage of mne functions.
+
+import torch
+
+epochs = train_set.datasets[0].windows
+epochs_tr = epochs.copy()
+
+X = torch.Tensor(epochs_tr._data)
+X_tr = transform.forward(X)
+epochs_tr._data = X_tr
```

Comment:
```suggestion
epochs_tr.apply_function(lambda x: transform.forward(torch.Tensor(x)).numpy())
```
Would this work @agramfort ?

Reply:
`forward` should not be called. Just transform(x) should work

Reply:
ah right yes

Reply:
Yes it should